File: CS/WebSite/DefaultDataSources.ascx.cs

using System;
using System.Web.UI;
using System.Web.UI.WebControls;
using DevExpress.Web.ASPxScheduler;

public partial class DefaultDataSources : System.Web.UI.UserControl {
	public ObjectDataSource AppointmentDataSource { get { return innerAppointmentDataSource; } }
	public ObjectDataSource ResourceDataSource { get { return innerResourceDataSource; } }

	public void AttachTo(ASPxScheduler control) {
		control.ResourceDataSource = this.ResourceDataSource;
		control.AppointmentDataSource = this.AppointmentDataSource;
		control.DataBind();
	}

	#region Site Mode implementation
	protected void innerAppointmentDataSource_ObjectCreated(object sender, ObjectDataSourceEventArgs e) {
		e.ObjectInstance = new CustomEventDataSource();
	}
	protected void innerResourceDataSource_ObjectCreated(object sender, ObjectDataSourceEventArgs e) {
		e.ObjectInstance = new CustomResourceDataSource();
	}
	#endregion
}

File: VB/WebSite/DefaultDataSources.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports DevExpress.Web.ASPxScheduler

Partial Public Class DefaultDataSources
	Inherits System.Web.UI.UserControl
	Public ReadOnly Property AppointmentDataSource() As ObjectDataSource
		Get
			Return innerAppointmentDataSource
		End Get
	End Property
	Public ReadOnly Property ResourceDataSource() As ObjectDataSource
		Get
			Return innerResourceDataSource
		End Get
	End Property

	Public Sub AttachTo(ByVal control As ASPxScheduler)
		control.ResourceDataSource = Me.ResourceDataSource
		control.AppointmentDataSource = Me.AppointmentDataSource
		control.DataBind()
	End Sub

	#Region "Site Mode implementation"
	Protected Sub innerAppointmentDataSource_ObjectCreated(ByVal sender As Object, ByVal e As ObjectDataSourceEventArgs)
		e.ObjectInstance = New CustomEventDataSource()
	End Sub
	Protected Sub innerResourceDataSource_ObjectCreated(ByVal sender As Object, ByVal e As ObjectDataSourceEventArgs)
		e.ObjectInstance = New CustomResourceDataSource()
	End Sub
	#End Region
End Class
